### ISSUE TITLE
Generated code produces compile error when accessing "data" dictionary in the "InputDict" struct if the name of the accessed property is "hash"

### ISSUE THREAD

**Issue by Anteo95** [apollographql/apollo-ios#2603]
## Bug report

Currently there is a compiler error in generated code when accessing "data" dictionary in the "InputDict" struct if the name of the accessed property is "hash".

## Versions

- `apollo-ios` SDK version: 1.0.2
- Xcode version: 14.1 RC
- Swift version: 5.7.1

## Steps to reproduce

1. execute generate code command.

## Further details
Here is a screenshot of errors produced by the compiler:

<img width="1196" alt="Screenshot 2022-10-24 at 15 38 55" src="https://user-images.githubusercontent.com/15553871/197539720-eff3c19a-c043-4e14-b7b9-9f51217f4165.png">

I think the reason why this fails is because InputDict has a function:

```swift
  public func hash(into hasher: inout Hasher) {
    hasher.combine(data._jsonEncodableValue?._jsonValue)
  }
```

and that causes the compiler to not be able to find "hash" field through dynamic member lookup.

changing generated code to this removes the compile error:

<img width="1184" alt="Screenshot 2022-10-24 at 15 44 13" src="https://user-images.githubusercontent.com/15553871/197540494-3a7f94dd-6260-4b3a-8ca7-82c9521efa3f.png">

If there is need for any additional info for solving this issue I will try to answer it here.

**Comment by calvincestari** [comment]
Hi @Anteo95 👋🏻 - thanks for the bug report. I'll do some more investigation but this looks like another parameter name we'll need to escape or mark as reserved.

**Comment by calvincestari** [comment]
@Anteo95 thanks for your direction on the solution. I tried many different things yesterday and it looks like that is the only viable one to fix it completely. It's particularly bothersome that the `hash` function is not something internal that we could simply rename.

**Comment by Anteo95** [comment]
@calvincestari no problem, glad i could help!

**Comment by calvincestari** [comment]
This issue is fixed with the merging of #2607 and will be included with the `1.0.3` release soon.

@Anteo95 I ended up simply removing `@dynamicMemberLookup`, which made the solution far easier.
